src_c/imageext.c

```diff
@@ -207,6 +208,22 @@ write_png(const char *file_name, SDL_RWops *rw, png_bytep *rows, int w, int h,
                  PNG_INTERLACE_NONE, PNG_COMPRESSION_TYPE_BASE,
                  PNG_FILTER_TYPE_BASE);
 
+    if (palette) {
+        doing = "set pallete";
+        const int ncolors = palette->ncolors;
+        int i;
+        if (!(color_ptr =
+                  (png_colorp)SDL_malloc(sizeof(png_colorp) * ncolors)))
+            goto fail;
+        for (i = 0; i < ncolors; i++) {
+            color_ptr[i].red = palette->colors[i].r;
+            color_ptr[i].green = palette->colors[i].g;
+            color_ptr[i].blue = palette->colors[i].b;
+        }
+        png_set_PLTE(png_ptr, info_ptr, color_ptr, ncolors);
+        free(color_ptr);
```

Comment:
This should use SDL_free because it was malloc'd with SDL_malloc. It could be that a malloc implementation is not exactly compatible on all platforms/configurations. So we try to match up python/SDL/libc uses when we can.